### ISSUE TITLE
using root as backup user

### ISSUE THREAD

**Issue by 3c2b2ff5** [elkarbackup/elkarbackup#382]
Hi,

I am using signed ssh public keys for authentication. So basically the user's public key and the hosts keys are signed by the sam CA. This way I don't need to `ssh-copy-id` to all my hosts.
I am wondering if it is possible to use `root` as my default backup user instead of `elkarbackup`?

Thanks

**Comment by elacunza** [comment]
Hi,

El 30/8/19 a las 0:40, 3c2b2ff5 escribió:
>
> Hi,
>
> I am using signed ssh public keys for authentication. So basically the 
> user's public key and the hosts keys are signed by the sam CA. This 
> way I don't need to |ssh-copy-id| to all my hosts.
> I am wondering if it is possible to use |root| as my default backup 
> user instead of |elkarbackup|?
>

I think you'll suffer permission issues. Why not copy root SSH private 
key to elkarbackup user?

Cheers

-- 
Zuzendari Teknikoa / Director Técnico
Binovo IT Human Project, S.L.
Telf. 943569206
Astigarraga bidea 2, 2º izq. oficina 11; 20180 Oiartzun (Gipuzkoa)
www.binovo.es

**Comment by 3c2b2ff5** [comment]
Unfortunately this won't work. There are three possibilities:

1. `ssh-copy-id root@client` or coping the public key by hand. This has the downside that it must repeated when ever reinstalling Elkarbackup. another way would be to save the `/var/lib/elkarbackup/.ssh` and replace it when needed. 
2. using signed ssh keys with the default backup user (elkarbackup) and create a `elkarbackup` user on each machine and give it sudo rights without password. So there is no need to `ssh-copy-id root@client` any more.
3. using signed ssh keys with root, since root exists on each linux machine and there is no need to `ssh-copy-id root@client` as well. 

The third option is much easier for my infrastructure and I would like to give it a try. So how can I set it up? which files need to be adjusted to make it possible?
Furthermore I am using puppet to deploy the Elkarbackup server and I did it successfully. I just need few more things to create/adjust with puppet. so where are the configuration files for the following:

1. set backup user to root
2. backup location
3. policies
4. pre and post scripts to backup Mysql server for instance.

Thank you very much.
